Automattic/wp-calypso
# Plans (state): Update plan-thank-you-card to utilise data-stores

Related to https://github.com/Automattic/wp-calypso/issues/86638

## Proposed Changes

Updates the plan-thank-you-card component to utilise Plans data-store for plan info
- updates `getCurrentPlan` with `Plans.useCurrentPlan`
- updates `getPlans` with `Plans.usePlans`
- updates `getTitle()` call with `plan.productNameShort` attached from the API/endpoint side on the plan objects
- converts from class to function form

### Caveats

Double check if this is ok:

> updates `getTitle()` call with `plan.productNameShort` attached from the API/endpoint side on the plan objects

## Why are these changes being made?


Reduce state duplication across local Calypso state and the packaged data-stores.

## Testing Instructions



Some checkout URLs to test. Smoke test each one -- they should look the same before and after this change:

- `/checkout/thank-you/no-site`
- `/thank-you/example.com/unknown-receipt` (where example.com should be changed to a real site hostname.)
- The thank-you page for a domain-only purchase (begin the purchase process at /domains).
- The thank-you page for a plan (begin the purchase process at /plans).

To test the actual component:
- /checkout/thank-you/no-site/:receiptId  ( use a receiptId from a domain transfer purchase )

## Pre-merge Checklist
